### ISSUE TITLE
creating generic utils with `InstantEntity` surfaces typing issue

### ISSUE THREAD

**Issue by uriva** [instantdb/instant#493]
```ts
export const db = init_experimental({
    adminToken: "...,
    appId: "...",
    schema: i.graph({ users: { email: i.string() } }),
});

type Collection = keyof typeof db.tx;

type Entity<T extends Collection> = InstantEntity<typeof db, T, {}>;

export const newEntity = async <T extends Collection>(
    type: T,
    props: Omit<Entity<T>, "id">,
) => {
    const theId = id();
    await db.transact(db.tx[type][theId].update(props));
    return theId;
};

// seems good
const alice = await newEntity("users", {    email: "alice@gmail.com",});

// Should be a typing error but is not
const bob = await newEntity("users", {    blabla: "bob@gmail.com",});

// Should be a typing error but is not
const eve = await newEntity("users", {  email: 123,});
```

**Comment by uriva** [comment]
also I wanted to add links parameter, but was not sure how to type it.

**Comment by stopachka** [comment]
Hey @uriva , I just published `0.17.3`, with a new utility type, which could help here: 

Here's a complete example: 

```typescript
import { i, id, init, UpdateParams } from "@instantdb/admin";

const _schema = i.schema({ entities: { users: i.entity({ email: i.string() }) } });
type _AppSchema = typeof _schema; 
interface AppSchema extends _AppSchema {} 
const schema: AppSchema = _schema; 

const db = init({
  adminToken: "...",
  appId: "...",
  schema,
});

type Collection = keyof typeof schema.entities;

type EntityUpdate<T extends Collection> = UpdateParams<typeof schema, T>;

export const newEntity = async <T extends Collection>(
  type: T,
  props: EntityUpdate<T>,
) => {
  const theId = id();
  await db.transact(db.tx[type][theId].update(props));
  return theId;
};

// seems good
const existing_attr_works = await newEntity("users", { email: "alice@gmail.com" });

// @ts-expect-error
const non_existing_attr_errors = await newEntity("users", { blabla: "bob@gmail.com" });

// @ts-expect-error
const wrong_type_errors = await newEntity("users", { email: 123 });

// to silence ts warnings
existing_attr_works;
non_existing_attr_errors;
wrong_type_errors;
```

After looking, I realized that InstaQLEntity doesn't fully satisfy `update`. I went ahead and expose the `update` utility type. I also exposed a `LinkParams` utility type, if you'd like to do the same for links. Let me know how it goes!

**Comment by uriva** [comment]
lgtm
